tests/system_test.sh

```diff
@@ -6,48 +6,82 @@
 #
 # Needs to be run from the root directory of the project.
 
+# Use `begin_fold` and `end_fold` to fold large pieces of output in the job log in Travis CI
+begin_fold() {
+    if [ "$TRAVIS" == true ]; then
+        echo "travis_fold:start:$1"
+    fi
+    echo -e "\e[33;1m$2\e[0m"
+}
 
+end_fold() {
+    if [ "$TRAVIS" == true ]; then
+        echo "travis_fold:end:$1"
+    fi
+}
 
-echo "travis_fold:start:docker-build"
-echo -e "\e[33;1mBuild docker images\e[0m"
+begin_fold docker-build "Build docker images"
 docker-compose build || { printf ">>> ERROR: Docker build failed"; exit 2; }
-echo "travis_fold:end:docker-build"
+end_fold docker-build
 
-echo "travis_fold:start:docker-up"
-echo -e "\e[33;1mStarting containers\e[0m"
-docker-compose up -d || { printf ">>> ERROR: Launching containers failed"; exit 2; }
-echo # `echo` to work around formatting issues after `docker-compose`
-echo "travis_fold:end:docker-up"
+begin_fold docker-up "Starting containers"
+# `--no-ansi` fixes the console formatting issue of `docker-compose`
+docker-compose --no-ansi up -d || { printf ">>> ERROR: Launching containers failed"; exit 2; }
+end_fold docker-up
+
+CARBON_HOST=127.0.0.1
+CARBON_PORT=2003
+CARBON_RETRIES=5
+CARBON_SLEEP_SEC=1
 
 # This is guestimate for the time when the system goes up and operational.
 # Doing this other ways is much more complex. May need to be increased on slow boxes.
-sleep 5
+for i in `seq 1 $CARBON_RETRIES`; do
+    nc $CARBON_HOST $CARBON_PORT
+    if [ $? -eq 0 ]; then
+        break
+    fi
+
```

Comment:
I don't see a wait around here. I expected to see a wait for `CARBON_SLEEP_SEC`.

Reply:
true, I indeed lost a sleep in there.